File: 0907/ABC267_C.py
N, M = map(int, input().split())
A = list(map(int, input().split()))
num = 0
for i in range(M):
    num += A[-(i+1)] * (M - i)
max_num = num
add_sum = sum(A[N-M:N])
for i in range(N-M):
    num -= M * A[-(i+1)]
    add_sum += A[N-i-M-1] - A[N-i-1]
    num += add_sum
    if max_num < num:
        max_num = num
print(max_num)

File: 0907/ABC267_C/ABC267_C.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ABC267C
{
    class Program
    {
        static void Main(string[] args)
        {
            var S = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = S[0];
            var M = S[1];
            var A = Console.ReadLine().Split(' ').Select(long.Parse).ToList();
            var init_num = A.Where((x, index) => index < M - 1).Select((x, index) => x * (index + 2)).Sum();
            var init_sub_sum = A.Where((x, index) => index < M - 1).Select(x => x).Sum();
            long num = init_num;
            long sub_sum = init_sub_sum;
            var solve = A.Where((x, index) => index <= N - M).Select((x, index) => {
                num += M * A[index + M - 1];
                num -= init_sub_sum;
                init_sub_sum += A[index + M - 1];
                if (index != 0) init_sub_sum -= A[index - 1];
                return num;
            });
            Console.WriteLine(solve.Max());
        }
    }
}